lib/semantics/check-do-concurrent.cc

```diff
@@ -204,6 +249,20 @@ class DoConcurrentLabelEnforce {
   void Post(const parser::ErrLabel &errLabel) { checkLabelUse(errLabel.v); }
   void Post(const parser::EndLabel &endLabel) { checkLabelUse(endLabel.v); }
   void Post(const parser::EorLabel &eorLabel) { checkLabelUse(eorLabel.v); }
+  void Post(const parser::DoConstruct &d) { do_depth_--; }
+  void checkName(const std::optional<parser::Name> &nm) {
+    if (!nm.has_value()) {
+      if (do_depth_ == 0) {
+        messages_.Say(currentStatementSourcePosition_,
+            "exit from DO CONCURRENT construct"_err_en_US);
+      }
+      return;
+    }
+    if (names_.find(nm.value().ToString()) == names_.end()) {
+      messages_.Say(currentStatementSourcePosition_,
+          "exit from DO CONCURRENT construct"_err_en_US);
```

Comment:
You might consider adding the name of the DO construct to the error message.

Reply:
I have added the name of the do concurrent construct and also the name of the exit target in the error message. And this simplifies and increases the accuracy of testing. Thanks for the suggestion.